packages/pwa-kit-create-app/scripts/create-mobify-app.js

```diff
@@ -861,47 +927,100 @@ const runGenerator = (context, {outputDir, templateVersion, verbose}) => {
                 })
         }
 
-        // Process selected Application Extensions
-        processAppExtensions(selectedAppExtensions, extractAppExtensions, appExtensionsDir)
+        // Check project type and handle appropriately
+        if (answers.project.type === 'appExtensionProject') {
+            const devOutputDir = p.join(outputDir, APP_EXTENSIBILITY_TYPESCRIPT_MINIMAL_DEV_DIR)
+
+            // Update the root package.json to add a start script
+            updatePackageJson(p.resolve(outputDir, 'package.json'), {
+                scripts: {
+                    start: 'npm --prefix ./dev start'
+                }
+            })
+
+            // Recursively call runGenerator for the 'typescript-minimal' preset
+            const extensionContext = {
+                ...context,
+                preset: {
+                    id: 'typescript-minimal',
+                    templateSource: {type: TEMPLATE_SOURCE_BUNDLE, id: 'typescript-minimal'},
+                    private: true
+                },
+                answers: {project: {type: 'PWAKitProject', name: 'typescript-dev'}}
+            }
+
+            await runGenerator(extensionContext, {
+                outputDir: devOutputDir,
+                templateVersion,
+                verbose,
+                isRecursiveCall: true
+            })
+
+            // Update the typescript-minimal dev package.json with dependencies
+            updatePackageJson(p.resolve(devOutputDir, 'package.json'), {
+                devDependencies: {[answers.project.name]: 'file:../'},
+                mobify: {app: {extensions: [answers.project.name]}}
+            })
 
-        // Update the generated project's package.json. NOTE: For bootstrapped projects this
-        // can be done in the template building. But since we have two types of project builds,
-        // (bootstrap/bundle) we'll do it here where it works in both scenarios.
-        const pkgJsonPath = p.resolve(outputDir, 'package.json')
-        const pkgJSON = readJson(pkgJsonPath)
+            // Create the .npmignore file, excluding the typescript-minimal dev folder
+            createNpmIgnoreFile(outputDir, [`${APP_EXTENSIBILITY_TYPESCRIPT_MINIMAL_DEV_DIR}/`])
+
+            //TODO: Avoid duplicated console.log Installing dependencies in terminal
```

Comment:
Is this comment about being concerned that we see installing of dependencies for the extension project and also for the "dev/example" project? I don't really thing that is a concern to be worried about.